### PR TITLE
Move Implicit Unique Constraint Creation from Migrate to Datamodel Parser

### PR DESCRIPTION
**Pull request by do4gr**
Moves the creation of implicit unique constraints for 1:1 relations into the standardization step and out of migrate.

### CODE REVIEW

File: migration-engine/migration-engine-tests/tests/schema_push/mod.rs [hunk 1]
```diff
@@ -257,3 +257,35 @@ fn sqlite_reserved_name_space_can_be_used(api: TestApi) {
...
+//               user   User @relation(fields: [userId], references: [id])
+//               userId Int       @map("user_id")
+//             }
+//      "#;
+//
+//     api.schema_push_w_datasource(plain_dm).send().assert_green();
+//     api.assert_schema().assert_table("A", |table| {
+//         table.assert_has_index_name_and_type("sqlite_unique", true);
+//         table.assert_has_index_name_and_type("sqlite_compound_unique", true);
+//         table.assert_has_index_name_and_type("sqlite_index", false)
```
Comment (tomhoule) [inline migration-engine/migration-engine-tests/tests/schema_push/mod.rs:289]: Is this test meant to be uncommented?

File: libs/datamodel/core/tests/attributes/relations/relations_positive.rs [hunk 2]
```diff
@@ -351,6 +351,90 @@ fn implicit_fk_name_definition_with_mapped_models_and_fields_other_order() {
...
+        .assert_relation_fk_name(Some("Post_user_id_1_user_id_2_fkey".to_string()));
+
+    schema.assert_has_model("Post").assert_has_index(IndexDefinition {
+        name: None,
+        db_name: Some("Post_user_id_1_user_id_2_key".to_string()),
+        fields: vec!["user_id_1".to_string(), "user_id_2".to_string()],
+        tpe: IndexType::Unique,
+        defined_on_field: false,
+    });
+}
```
Comment (tomhoule) [inline libs/datamodel/core/tests/attributes/relations/relations_positive.rs:436]: :+1:

**Review by tomhoule — COMMENTED** [review]
Better :+1:

File: migration-engine/migration-engine-tests/tests/schema_push/mod.rs (hunk 1 above)
Comment (do4gr) [reply migration-engine/migration-engine-tests/tests/schema_push/mod.rs:289]: I decided to do the refactoring in a first PR before doing the name validation in a second one. But I already had set up a failing test for the follow up PR so I decided to just comment it out for now.

**Pull request merged**